datumaro/plugins/yolo_format/extractor.py

```diff
@@ -186,9 +188,12 @@ def _load_categories(names_path):
 
         return label_categories
 
-    def __iter__(self):
-        for subset in self._subsets.values():
-            for item in subset:
+    @scoped(arg_name='scope') # pylint: disable=no-value-for-parameter
+    def __iter__(self, *, scope: Scope = None):
```

Comment:
Useless default argument?

Reply:
It just suppresses the linter message, because iter cannot have arguments. But we need an argument here (unless we go for code generation), because otherwise it is not being captured by a generator (which appears from yield) (see #661).

Reply:
After experimenting and thinking over, I decided to remove closing from plugins, but restore `finish`. In the end of the day, it feels natural for pbar (and ui, possible) to manage its children objects. Finishing is still important for signaling the finish of the loading in a pbar. In case of error, we aren't going to have any new pbars. There is still a question if we need a dedicated close (to close an unfinished pbar), but it is not that important now.